pages/cookbook/jettons.mdx

```diff
@@ -1,9 +1,161 @@
 # Fungible Tokens (Jettons)
 
-import { Callout } from 'nextra/components'
+import { Callout } from 'nextra-theme-docs';
 
-<Callout emoji="🚨">
+This page lists common examples of working with [jettons](https://docs.ton.org/develop/dapps/asset-processing/jettons).
 
-  This page is a stub. [Contributions are welcome!](https://github.com/tact-lang/tact-docs/issues)
+## Accepting jetton transfer
+
+Transfer notification message have the following structure.
+
+```tact
+message(0x7362d09c) TransferNotification {
+    queryId: Int as uint64;
+    jettonAmount: Int as coins;
+    fromAddress: Address;
+    forwardPayload: Slice as remaining;
+}
+```
+
+Use [receiver](/book/receive) function to accept token notification message.
+
+<Callout type="warning" emoji="⚠️">
+
+  Sender of transfer notification must be validated!
+
+</Callout>
+
+Validation can be done using jetton wallet state init and calculating jetton address.
+Note, that notifications are coming from YOUR contract's jetton wallet, so [`myAddress()`](/ref/core-common#myaddress) should be used in owner address field.
+Wallet initial data layout is shown below, but sometimes it can differ.
+
+```tact
+struct JettonWalletData {
+    balance: Int as coins;
+    ownerAddress: Address;
+    jettonMasterAddress: Address;
+    jettonWalletCode: Cell;
+}
+
+fun calculateJettonWalletAddress(ownerAddress: Address, jettonMasterAddress: Address, jettonWalletCode: Cell): Address {
+    let initData = JettonWalletData{
+        balance: 0,
+        ownerAddress,
+        jettonMasterAddress,
+        jettonWalletCode,
+    };
+
+    return contractAddress(StateInit{code: jettonWalletCode, data: initData.toCell()});
+}
+
+contract Sample {
+    jettonWalletCode: Cell;
+    jettonMasterAddress: Address;
+
+    init(jettonWalletCode: Cell, jettonMasterAddress: Address) {
+        self.jettonWalletCode = jettonWalletCode;
+        self.jettonMasterAddress = jettonMasterAddress;
+    }
+
+    receive(msg: TransferNotification) {
+        let myJettonWalletAddress = calculateJettonWalletAddress(myAddress(), self.jettonMasterAddress, self.jettonWalletCode);
+        require(sender() == myJettonWalletAddress, "Notification not from your jetton wallet!");
+
+        // your logic of processing token notification
+    }
+}
+```
+
+## Sending jetton transfer
+
+To send jetton transfer use [`send(){:tact}`](/book/send) function.
+If you want to have full control over the sending process and require some complex logic for it,
+use [`nativeSendMessage(){:tact}`](/ref/core-advanced#nativesendmessage) function.
```

Comment:
shouldn't we recommend using `send()` here instead of `nativeSendMessage()`?

Reply:
Removed completely this paragraph, User will follow nativeSendMessage in /book/send if he realy needs to